Src/Library/Extensions/ReflectionExtensions.cs

```diff
@@ -126,10 +126,23 @@ expression.Body switch
         return Expression.Lambda<Func<object?, (bool, object)>>(
             block,
             inputParameter
-            ).Compile();
+        ).Compile();
     }
 
-    private static object? DeserializeArrayString(object? input, Type tProp)
+    private static object? DeserializeJsonObjectString(object? input, Type tProp)

```

Comment:
@vpetrusevici let's just do this and call it a day. this will also work for other binding sources such as form fields, header values, etc.

Reply:
it will work if we send query like ?x={"name":"x","age":24}.
But I want to implement a behavior from mvc [FromQuery]. Query should be ?name=x&age=24

Reply:
i suggest you implement it in a custom model binder as i feel this is not something i wanna add to the library itself. there must be a really good reason why it's not supported by minimal apis also.